### PR TITLE
fix: camelize handle django lazy strings

### PR DESCRIPTION
**Pull request by davecoates**

### CODE REVIEW

File: allianceutils/util/camel_case.py [hunk 1]
```diff
@@ -171,7 +176,9 @@ def _transform_data(data, transform_key: Callable, ignore_lookup: Dict):
     # containing File objects for any file fields.
     # At least for now we don't support numeric indices in ignores, so '*' is the only ignore lookup index
     # that can match a list/iterable
-    if isinstance(data, Iterable) and not isinstance(data, (str, bytes, File)):
+    # `Promise` is included here as django uses that as the base for prozy class created in lazy functions, eg.
```
Comment (vbudovski-alliance) [inline allianceutils/util/camel_case.py:179]: s/prozy/proxy

**Comment by vbudovski-alliance** [comment]
LGTM

**Comment by levic** [comment]
@davecoates can you confirm my understanding:

This came up because field labels are lazy in order to defer localization. This won't properly handle eg a lazy function that might return an int because we'll cast it to a str when we see that it's lazy. This is arguably 'wrong', but we don't claim to support general lazy expressions and in any event is no worse than what DjangoJSONEncoder does: it's the caller's responsibility to evaluate those lazy expressions ahead of time.

LGTM if that above understanding is correct.

**Comment by davecoates** [comment]
> @davecoates can you confirm my understanding:
> 
> This came up because field labels are lazy in order to defer localization. This won't properly handle eg a lazy function that might return an int because we'll cast it to a str when we see that it's lazy. This is arguably 'wrong', but we don't claim to support general lazy expressions and in any event is no worse than what DjangoJSONEncoder does: it's the caller's responsibility to evaluate those lazy expressions ahead of time.
> 
> LGTM if that above understanding is correct.

yep that's correct

**Pull request closed without merging**